apps/api/src/controllers/collaboratorsController.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { getDbInstance } from '@/db/index.js';
+import { applicationSvc } from '@/service/applicationService.js';
+import { collaboratorsSvc } from '@/service/collaboratorsService.js';
+import { type ApplicationService, type CollaboratorModel, type CollaboratorsService } from '@/service/types.js';
+import { failure } from '@/utils/results.js';
+import { CollaboratorDTO } from '@pcgl-daco/data-model';
+
+/* Type to assert Collaborator record has valid required fields */
+interface ValidCollaboratorDTO extends CollaboratorDTO {
+	collaboratorFirstName: string;
+	collaboratorLastName: string;
+	collaboratorPositionTitle: string;
+	collaboratorInstitutionalEmail: string;
+}
+
+/**
+ * Creates a new collaborator and returns the created data.
+ * @param application_id - ID of related application record to associate with Collaborators
+ * @param user_id - ID of Applicant updating the application
+ * @param collaborators - Array of new Collaborators to create
+ * @returns Success with Collaborator data array / Failure with Error.
+ */
+export const createCollaborators = async ({
+	application_id,
+	user_id,
+	collaborators,
+}: {
+	application_id: number;
+	user_id: string;
+	collaborators: CollaboratorDTO[];
+}) => {
+	const database = getDbInstance();
+	const collaboratorsRepo: CollaboratorsService = collaboratorsSvc(database);
+	const applicationRepo: ApplicationService = applicationSvc(database);
+
+	const applicationResult = await applicationRepo.getApplicationById({ id: application_id });
```

Comment:
Before fetching the app we should perform our checks if this function should run, ie. do the auth check first. (I know auth is a TODO atm but lets put it in the right place).

Reply:
This is confirming that the application id and user id match, and you can only do that by comparing what's in the request body with what's in the database

So I've built it to retrieve a record, confirm the request details match; if they don't match you error here; if they do match you proceed with the rest of the application actions

This was a solution to putting in 'working' auth until we have a different approach defined
So I could move this check but it won't actually be checking the request body against 'real' data

Reply:
So -- there's one layer of auth validation on the router (is userId included). 
Then this line in the controller adds a second layer of validation with the db (userId and applicationId are a match). Both are to be replaced when real Auth is added.

Let me know whether to keep both, or reorganize, or what you're thinking